lib/tasks/backfill_new_jsonb_columns.rake

```diff
@@ -1,91 +1,26 @@
-desc "Migrate access_limits json columns"
-task access_limits: :environment do
-  puts "starting access limits"
-  acc_lim_sql = "UPDATE access_limits SET temp_users = users, temp_organisations = organisations;"
-  ActiveRecord::Base.connection_pool.with_connection { |con| con.exec_query(acc_lim_sql) }
-end
-
-desc "Migrate editions json columns"
-task editions: :environment do
-  record = Edition.find_by(id: 704530)
-  details = record.details.to_json.gsub(/\\u0000/, "")
-  details = JSON.parse(details).symbolize_keys
-  record.update(details: details)
-
-  low = 0
-  high = 1000
-  total = Edition.maximum("id")
-
-  sql = "UPDATE editions SET temp_details = details, temp_routes = routes, " \
-        "temp_redirects = redirects WHERE id > #{low} AND id <= #{high};"
-
-  while high < total
-    ActiveRecord::Base.connection_pool.with_connection { |con| con.exec_query(sql) }
-    puts "finished editions batch #{low} to #{high}"
-    low += 1000
-    high += 1000
-  end
-end
-
-desc "Migrate events json columns"
-task events: :environment do
-  event = Event.find_by(id: 31148138)
-  payload = event.payload.to_json.gsub(/\\u0000/, "")
-  payload = JSON.parse(payload).symbolize_keys
-  event.update(payload: payload)
-
-  low = 0
-  high = 1000
-  total = Event.maximum("id")
-
-  sql = "UPDATE events SET temp_payload = payload WHERE id > #{low} AND id <= #{high};"
+desc "Backfill new jsonb columns with data from original json columns"
+task backfill_new_jsonb_columns: :environment do
+  sql = "UPDATE editions SET temp_details = details, temp_routes = routes, temp_redirects = redirects WHERE id in (SELECT id FROM editions WHERE temp_details IS NULL LIMIT 100);"
 
-  while high < total
+  num = (Edition.count / 100) + 1
+  num.times do |i|
     ActiveRecord::Base.connection_pool.with_connection { |con| con.exec_query(sql) }
```

Comment:
Are newly created / updated editions correct?  Is it only old editions which have the problem?

Reply:
It seems like all the new ones are fine